### PR TITLE
[Tracing] Remove dependence of tracing on jinja2

### PR DESCRIPTION
**Pull request by lumoslnt**
# Description

Since the promptflow-tracing package should only depend on openai and opentelemetry, we should remove the dependence of tracing on jinja2.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — DISMISSED** [review]

**Review by zhengfeiwang — DISMISSED** [review]

File: src/promptflow/promptflow/tracing/_utils.py [hunk 1]
```diff
@@ -59,18 +58,23 @@ def serialize(value: object, remove_null: bool = False, serialization_funcs: Dic
 
 
 def get_input_names_for_prompt_template(template_str):
-    input_names = []
-    env = Environment()
-    template = env.parse(template_str)
-    input_names.extend(sorted(meta.find_undeclared_variables(template), key=lambda x: template_str.find(x)))
+    try:
```
Comment (thy09) [inline src/promptflow/promptflow/tracing/_utils.py:61]: Add more documents about the try logic here

Comment (thy09) [reply src/promptflow/promptflow/tracing/_utils.py:61]: Maybe we can try except then directly return a empty list if ImportError.
So we don't need to update the other codes.

Comment (lumoslnt) [reply src/promptflow/promptflow/tracing/_utils.py:61]: Added.

File: src/promptflow/promptflow/tracing/_utils.py [hunk 2]
```diff
@@ -59,18 +58,25 @@ def serialize(value: object, remove_null: bool = False, serialization_funcs: Dic
...
+        # We need to parse jinja template only when the promptflow is installed and run flow with PromptTemplate
+        # type input, so using try-catch to avoid the dependency of jinja2 when it's not needed.
+        from jinja2 import Environment, meta
```
Comment (zhengfeiwang) [inline src/promptflow/promptflow/tracing/_utils.py:64]: not sure simply jinja2 or promptflow.core (I don't know the precise path...)?

if a user installs jinja2, but not from promptflow, is the below lines executed expected?

Comment (lumoslnt) [reply src/promptflow/promptflow/tracing/_utils.py:64]: 1. It should be jinja2.
2. We need to parse jinja, only if the user installed the promptflow and used PromptTemplate type as input.

**Review by zhengfeiwang — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**